frontend/src/components/Secret.js

```diff
@@ -0,0 +1,82 @@
+import React, { useEffect } from 'react'
+import { useSelector, useDispatch, batch } from 'react-redux'
+import { useHistory } from 'react-router-dom'
+
+import { API_URL } from '../reuseables/urls'
+
+import user from '../reducers/user'
+import { MainContainer, SubContainer, Header, Text, Anchor, Button } from './styled-components/secret-style'
+
+const Secret = () => {
+    const accessToken = useSelector(store => store.user.accessToken)
+    const secretMessage = useSelector(store => store.user.secretMessage)
+
+    const dispatch = useDispatch()
+    const history = useHistory()
+
+    useEffect(() => {
+      if (!accessToken) {
+        history.push('/login')
+      }
+    }, [accessToken, history])
+
+    useEffect(() => {
+      if (accessToken) {
+        const options = {
+          method: 'GET',
+          headers: {
+              Authorization: accessToken
+          }
+        }
+        fetch(API_URL('secret'), options)
+          .then(res => res.json())
+          .then(data => {
+            if (data.success) {
+              batch(() => {
+                dispatch(user.actions.setSecretMessage(data.secretMessage))
+                dispatch(user.actions.setErrors(null))
+              })
+            } else {
+              dispatch(user.actions.setErrors(data))
+            }
+          })
+      } 
+    }, [accessToken, dispatch]) 
```

Comment:
your second useEffect is only valid if token is present, so its logic should be moved inside the first one as the `else` block:

```jsx
useEffect(() => {
    if (!accessToken) {
      history.push('/login')
    } else {
      const options = { ... }
      ...
    }
  }, [accessToken, history, dispatch])
  ```